patches/installer/0007-Don-t-restore-WinForms-and-WPF-templates-for-source-.patch

```diff
@@ -0,0 +1,74 @@
+     <Bundled31Templates Include="Microsoft.DotNet.Web.ProjectTemplates.3.1" PackageVersion="$(AspNetCorePackageVersionFor31Templates)" />
+     <Bundled31Templates Include="Microsoft.DotNet.Web.Spa.ProjectTemplates.3.1" PackageVersion="$(AspNetCorePackageVersionFor31Templates)" />
+@@ -76,8 +72,6 @@
+     <Bundled30Templates Include="Microsoft.DotNet.Common.ItemTemplates" PackageVersion="$(MicrosoftDotNetCommonItemTemplates30PackageVersion)" />
+     <Bundled30Templates Include="Microsoft.DotNet.Common.ProjectTemplates.3.0" PackageVersion="$(MicrosoftDotNetCommonProjectTemplates30PackageVersion)" />
+     <Bundled30Templates Include="Microsoft.DotNet.Test.ProjectTemplates.3.0" PackageVersion="$(MicrosoftDotNetTestProjectTemplates30PackageVersion)" />
+-    <Bundled30Templates Include="Microsoft.Dotnet.Wpf.ProjectTemplates" PackageVersion="$(MicrosoftDotnetWpfProjectTemplates30PackageVersion)" />
+-    <Bundled30Templates Include="Microsoft.Dotnet.WinForms.ProjectTemplates" PackageVersion="$(MicrosoftDotnetWinFormsProjectTemplates30PackageVersion)" />
+     <Bundled30Templates Include="Microsoft.DotNet.Web.ItemTemplates" PackageVersion="$(AspNetCorePackageVersionFor30Templates)" />
+     <Bundled30Templates Include="Microsoft.DotNet.Web.ProjectTemplates.3.0" PackageVersion="$(AspNetCorePackageVersionFor30Templates)" />
+     <Bundled30Templates Include="Microsoft.DotNet.Web.Spa.ProjectTemplates.3.0" PackageVersion="$(AspNetCorePackageVersionFor30Templates)" />
+@@ -154,18 +148,6 @@
+                               [$(NUnit3Templates31PackageVersion)];
+                               [$(NUnit3Templates50PackageVersion)];
+                               " />
+-
+-   <PackageDownload Update="Microsoft.Dotnet.Wpf.ProjectTemplates"
+-                    Version="[$(MicrosoftDotnetWpfProjectTemplates30PackageVersion)];
+-                             [$(MicrosoftDotnetWpfProjectTemplates31PackageVersion)];
+-                             [$(MicrosoftDotnetWpfProjectTemplates50PackageVersion)];
+-                             " />
+-
+-    <PackageDownload Update="Microsoft.Dotnet.Winforms.ProjectTemplates"
+-                     Version="[$(MicrosoftDotnetWinFormsProjectTemplates30PackageVersion)];
+-                              [$(MicrosoftDotnetWinFormsProjectTemplates31PackageVersion)];
+-                              [$(MicrosoftDotnetWinFormsProjectTemplates50PackageVersion)];
+-                             " />
+   </ItemGroup>
+ 
+   <Target Name="LayoutTemplates"
+@@ -173,10 +155,6 @@
+ 
+   <Target Name="LayoutTemplatesForSDK"
+           DependsOnTargets="SetupBundledComponents;CalculateTemplatesVersions">
+-    <ItemGroup Condition="!$(ProductMonikerRid.StartsWith('win'))">
+-      <Bundled50Templates Remove="Microsoft.Dotnet.Wpf.ProjectTemplates" />
+-      <Bundled50Templates Remove="Microsoft.Dotnet.WinForms.ProjectTemplates" />
```

Comment:
To make this patch more mergeable, we could have a similar condition on the packagedownloads. To save time now, this seems fine though.

Reply:
Logged https://github.com/dotnet/source-build/issues/1802 for this.